### DISCUSSION TITLE
How to call an async function inside update?

### DISCUSSION THREAD

**Discussion by askasp** [davidpdrsn/axum-live-view#154] [Q&A] [answered]
I saw this issue https://github.com/davidpdrsn/axum-live-view/issues/144

And I assume it is possible to somehow write to a database within the update function. However, this "solution" causes a runtime within a runtime error (which makes sense).

![image](https://user-images.githubusercontent.com/41793839/210111780-fcf4d8a2-e35e-4c98-8f3e-a2ca9c553090.png)

If you have time i would really appreciate an example on how you would write to a database or do any other async call from within the update function. 

Happy holidays :)

**Comment by davidpdrsn** [comment, marked as answer]
`update` is intentionally not `async` as that would make it too easy to block the view from rendering, making the experience slow.

Instead use `Updated::spawn` to run a future in the background and send a message back to the view when it completes, so you can re-render.

**Reply by askasp** [reply]
Thanks!
